File: GenZCoders-main/Backend/GenZCoders/Models/CourseRound.cs
using System;
using System.Collections.Generic;

namespace GenZCoders.Models;

public partial class CourseRound
{
    public long Id { get; set; }

    public long CourseId { get; set; }

    public decimal? RoundNumber { get; set; }

    public DateTime? StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public long? MaxStudents { get; set; }

    public long StatusId { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? Question1 { get; set; }

    public string? Question2 { get; set; }

    public string? Question3 { get; set; }

    public string? Question4 { get; set; }

    public string? Question5 { get; set; }

    public string? Question6 { get; set; }

    public string? Question7 { get; set; }

    public string? Question8 { get; set; }

    public string? Question9 { get; set; }

    public string? Question10 { get; set; }

    public long? MinStudents { get; set; }

    public decimal? Price { get; set; }

    public int? AutomatedWorkFlowJump { get; set; }

    public long? CourseRoundGroupId { get; set; }

    public long? TrialStatusID { get; set; }

    public virtual Course Course { get; set; } = null!;

    public virtual Status Status { get; set; } = null!;

    public virtual Status? TrialStatus { get; set; }

    public virtual ICollection<CourseMaterial> CourseMaterials { get; set; } = new List<CourseMaterial>();

    public virtual ICollection<CourseRoundInstructor> CourseRoundInstructors { get; set; } = new List<CourseRoundInstructor>();

    public virtual ICollection<Application> Applications { get; set; } = new List<Application>();

    public virtual CourseRound? CourseRoundGroup { get; set; }

    public virtual ICollection<CourseRound> GroupedRounds { get; set; } = new List<CourseRound>();
    public virtual ICollection<ExamQuestionBank> ExamQuestionBanks { get; set; } = new List<ExamQuestionBank>();
}

File: GenZCoders-main/Frontend/src/api/models/course-round.ts
export interface CourseRoundDto {
  id: number;
  roundNumber: number;
  startDate: string; // DateOnly format: YYYY-MM-DD
  endDate: string; // DateOnly format: YYYY-MM-DD
  minStudents?: number | null;
  maxStudents?: number | null;
  /** Paid (enrolled) students in this round, computed by the server. */
  enrolledCount?: number;
  price?: number | null;
  courseId: number;
  courseName?: string;
  statusId?: number;
  status?: string;
  courseRoundGroupId?: number | null;
  groups?: CourseRoundGroupItemDto[];
  question1?: string | null;
  question2?: string | null;
  question3?: string | null;
  question4?: string | null;
  question5?: string | null;
  question6?: string | null;
  question7?: string | null;
  question8?: string | null;
  question9?: string | null;
  question10?: string | null;
  createdAt?: string;
  mainInstructorId?: number;
  instructorId?: number;
  instructorName?: string;
  InstructorId?: number;
  instructor?: string;
  Instructor?: string;
  weekTitles?: string[];
}

export interface CourseRoundGroupItemDto {
  id: number;
  roundNumber: number;
  startDate?: string;
  endDate?: string;
  price?: number | null;
}

export interface CourseRoundDetailsDto {
  id: number;
  roundNumber: number;
  courseName: string;
  courseId?: number | null;
  startDate?: string;
  endDate?: string;
  price?: number | null;
  courseRoundGroupId?: number | null;
  minStudents?: number | null;
  maxStudents?: number | null;
  statusId?: number;
  instructorId?: number | null;
  groups?: CourseRoundGroupItemDto[];
  weekTitles?: string[];
  instructorName?: string;
  question1?: string | null;
  question2?: string | null;
  question3?: string | null;
  question4?: string | null;
  question5?: string | null;
  question6?: string | null;
  question7?: string | null;
  question8?: string | null;
  question9?: string | null;
  question10?: string | null;
  automatedWorkFlowJump?: number | null;
}

export interface CreateCourseRoundRequest {
  courseId: number;
  roundNumber: number;
  startDate: string; // DateOnly format: YYYY-MM-DD
  endDate: string; // DateOnly format: YYYY-MM-DD
  minStudents?: number | null;
  maxStudents?: number | null;
  price?: number | null;
  courseRoundGroupId?: number | null;
  question1?: string | null;
  question2?: string | null;
  question3?: string | null;
  question4?: string | null;
  question5?: string | null;
  question6?: string | null;
  question7?: string | null;
  question8?: string | null;
  question9?: string | null;
  question10?: string | null;
  automatedWorkFlowJump?: number | null;
}

export interface UpdateCourseRoundRequest {
  courseId?: number | null;
  roundNumber: number;
  startDate?: string; // DateOnly format: YYYY-MM-DD
  endDate?: string; // DateOnly format: YYYY-MM-DD
  minStudents?: number | null;
  maxStudents?: number | null;
  price?: number | null;
  courseRoundGroupId?: number | null;
  statusId: number;
  question1?: string | null;
  question2?: string | null;
  question3?: string | null;
  question4?: string | null;
  question5?: string | null;
  question6?: string | null;
  question7?: string | null;
  question8?: string | null;
  question9?: string | null;
  question10?: string | null;
}

export interface PatchCourseRoundRequest {
  courseId?: number | null;
  roundNumber?: number | null;
  startDate?: string;
  endDate?: string;
  minStudents?: number | null;
  maxStudents?: number | null;
  price?: number | null;
  statusId?: number | null;
  courseRoundGroupId?: number | null;
  question1?: string | null;
  question2?: string | null;
  question3?: string | null;
  question4?: string | null;
  question5?: string | null;
  question6?: string | null;
  question7?: string | null;
  question8?: string | null;
  question9?: string | null;
  question10?: string | null;
}
